core/templates/dev/head/components/state-directives/answer-group-editor/answer-group-editor.directive.ts

```diff
@@ -83,16 +83,22 @@ oppia.directive('answerGroupEditor', [
           ctrl.misconceptionName = null;
           ctrl.misconceptions = StateEditorService.getMisconceptions();
 
-          var _getTaggedMisconceptionName = function(misconceptionId) {
-            for (var i = 0; i < ctrl.misconceptions.length; i++) {
-              if (
-                ctrl.misconceptions[i].getId() === misconceptionId) {
-                ctrl.misconceptionName = ctrl.misconceptions[i].getName();
+          var _getTaggedMisconceptionName = function(skillMisconceptionId) {
+            if (skillMisconceptionId !== null) {
+              var skillId = skillMisconceptionId.split('-')[0];
```

Comment:
Verify that split() actually does give an array with 2 elements, or else you will get errors here and below

Reply:
Done.